dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/JdbcAnalyticsManager.java

```diff
@@ -205,6 +201,137 @@ public JdbcAnalyticsManager( QueryPlanner queryPlanner,
+                    if ( k.contains( coc.getUid() ) )
+                    {
+                        aggregatedKeyValueMap.put( k, keyValueMap.get( k ) );
+                    }
+                } );
+            }
+        } );
+        return aggregatedKeyValueMap;
+    }
+
+    private static Map<String, Object> getAggregatedCategoryOptionValues( Map<String, Object> keyValueMap,
+        String coc_uid,
+        String co_uid )
+    {
+        final short CATEGORY_OPTION_POSITION = 1;
+        final short MIN_SIZE_TYPICAL_FOR_EXPRESSION_WITH_COC = 3;
+        Map<String, Object> aggregatedKeyValueMap = new HashMap<>();
+
+        keyValueMap.forEach( ( k, v ) -> {
+            String[] tokens = k.split( "-" );
+            if ( tokens.length < MIN_SIZE_TYPICAL_FOR_EXPRESSION_WITH_COC
+                || !tokens[CATEGORY_OPTION_POSITION].equals( coc_uid ) )
+            {
+                return;
+            }
+            tokens[CATEGORY_OPTION_POSITION] = co_uid;
+            String newKey = String.join( "-", tokens );
+            putToAggregatedKeyValueMap( aggregatedKeyValueMap, v, newKey );
+        } );
+        removeTrailingCommasFromAggregatedStringValues( aggregatedKeyValueMap );
+        return aggregatedKeyValueMap;
+    }
+
+    private static void removeTrailingCommasFromAggregatedStringValues( Map<String, Object> aggregatedKeyValueMap )
+    {
+        aggregatedKeyValueMap.forEach( ( k, v ) -> {
+            if ( v instanceof String )
+            {
+                String value = (String) v;
+                if ( value.charAt( value.length() - 1 ) == ',' )
+                {
+                    aggregatedKeyValueMap.replace( k, StringUtils.substring( value, 0, value.length() - 1 ) );
+                }
+            }
+        } );
+    }
+
+    private static void putToAggregatedKeyValueMap( final Map<String, Object> aggregatedKeyValueMap, Object value,
+        String key )
+    {
+        AtomicReference<Double> doubleSum = new AtomicReference<>( 0.0 );
+        StringBuilder stringSum = new StringBuilder();
+
+        if ( value instanceof Double )
+        {
+            if ( aggregatedKeyValueMap.containsKey( key ) )
+            {
+                doubleSum.updateAndGet( v1 -> v1 + (Double) value + (Double) aggregatedKeyValueMap.get( key ) );
+            }
+            else
+            {
+                doubleSum.updateAndGet( v1 -> v1 + (Double) value );
+            }
+            aggregatedKeyValueMap.put( key, doubleSum.get() );
+
+        }
+        else if ( value instanceof String )
+        {
+            if ( aggregatedKeyValueMap.containsKey( key ) )
+            {
+                stringSum.append( value + "," + aggregatedKeyValueMap.get( key ) + "," );
+            }
+            else
+            {
+                stringSum.append( (String) value ).append( "," );
+            }
+            aggregatedKeyValueMap.put( key, stringSum );
+        }
+    }
```

Comment:
I see what this logic is doing. This is the first time we've had to sum the values from multiple query result rows for a `DataElementOperand`. I wish there were a simpler way, but I'm not seeing it yet.

Can we avoid the constants CATEGORY_OPTION_POSITION and MIN_SIZE_TYPICAL_FOR_EXPRESSION_WITH_COC? They seems fragile to me. (But it's much better to have explicit names as you have done, instead of just the numbers!)